### ISSUE TITLE
Saved system settings repeating

### ISSUE THREAD

**Issue by DazedxJoker** [svof/svof#412]
Everytime you change class 1 more iteration of Saved system settings is displayed when you QQ.

![image](https://user-images.githubusercontent.com/14912622/61094656-159b8a80-a41e-11e9-9ef8-dc7a34127f39.png)

**Comment by DazedxJoker** [comment]
So, we discovered a way to reliably cause the repeating Saved system settings. It seems to be showing up once per class change. Which is when the system reloads itself. I've narrowed it down to this line:

signals.quit:add_pre_emit(function () signals.saveconfig:emit() end)

If you execute 
vlua svo.signals.quit:add_pre_emit(function () svo.signals.saveconfig:emit() end)

 Then QQ it will show an additional Saved system settings for each time you execute it.

**Comment by vadi2** [comment]
This is one of the things I had to do - make event registration be 'safe' even when reloading. The fix for this was to patch LuaNotify to accept a name for the connection handler, for example:

```lua
signals.gmcpcharname:connect(function ()
  signals.enablegmcp:emit()
end, 'emit enablegmcp')
```

Here 'emit enablegmcp' will overwrite any previous function in its place, preventing duplicating.

I've already updated `Signal:connect` and `Signal:add_post_emit` to allow a name for the function, but not yet `Signal:add_pre_emit` - see how the changes were done in the other two, do the same to `Signal:add_pre_emit`, and update all `add_pre_emit` calls to have a name.
